src/SplitPackages/Program.cs

```diff
@@ -0,0 +1,288 @@
+{sourcePackages}";
+                throw new InvalidOperationException(msg);
+            }
+
+            if (!File.Exists(csvFile))
+            {
+                var msg = $@"csv file does not exist or is not a file
+{csvFile}";
+                throw new InvalidOperationException(msg);
+            }
+
+            if (!Directory.Exists(destinationFolder))
+            {
+                var msg = $@"Destination folder does not exist, creating folder at:
+{destinationFolder}";
+                WriteInformational(msg);
+                Directory.CreateDirectory(destinationFolder);
+            }
+
+            return new Arguments
+            {
+                DestinationFolder = destinationFolder,
+                CsvFile = csvFile,
+                SourceFolder = sourcePackages
+            };
+        }
+
+        private static IList<PackageItem> GetPackagesFromSourceFolder(string sourceFolder)
+        {
+            var packages = Directory.EnumerateFiles(sourceFolder);
+
+            return packages.Select(p => new PackageItem
+            {
+                OriginPath = p,
+                Name = Path.GetFileName(p),
+                FoundInFolder = true
+            }).ToList();
+        }
+
+        private static IList<PackageItem> GetPackagesFromCsvFile(string csvFile)
+        {
+            var lines = File.ReadAllLines(csvFile);
+            var parsedLines = lines.Select(l => l.Split(','));
+
+            return parsedLines.Select(parsed => new PackageItem
+            {
+                Name = parsed[0].Trim(),
+                DestinationFolderName = parsed[1].Trim(),
+                FoundInCsv = true
+            }).ToList();
+        }
+
+        private static IList<PackageItem> ExpandPackages(IList<PackageItem> packagesFromCsv, string sourceFolder)
+        {
+            var allPackages = new HashSet<PackageItem>();
+
+            for (int i = 0; i < packagesFromCsv.Count; i++)
+            {
+                var foundPackages = Directory.EnumerateFiles(sourceFolder, packagesFromCsv[i].Name);
+                var expandedPackages = foundPackages.Select(fp => new PackageItem
+                {
+                    OriginPath = fp,
+                    Name = Path.GetFileName(fp),
+                    FoundInCsv = true,
+                    FoundInFolder = true,
+                    DestinationFolderName = packagesFromCsv[i].DestinationFolderName
+                }).ToList();
+
+                foreach (var package in expandedPackages)
+                {
+                    allPackages.Add(package);
+                }
+
+                if (expandedPackages.Count > 0)
+                {
+                    packagesFromCsv[i].FoundInFolder = true;
+                }
+            }
+
+            return allPackages.ToList();
+        }
+
+        private static void EnsureNoExtraSourcePackagesFound(IList<PackageItem> source, IList<PackageItem> expanded)
```

Comment:
Still need to go through the list of expanded packages to mark them as found in source, so it's fine like it is right now